circleguard/utils.py

```diff
@@ -114,12 +114,15 @@ def __init__(self, checks, run_id, event):
         self.run_id = run_id
         self.event = event
 
-class Player():
-    def __init__(self, data, replay, username, mods, buffer, cursor_color, pos):
-        self.data = data
+
+class Player:
+    def __init__(self, replay, cursor_color):
         self.replay = replay
-        self.username = username
-        self.mods = mods
-        self.buffer = buffer
         self.cursor_color = cursor_color
-        self.pos = pos
+        self.username = replay.username
+        self.t = replay.t
+        self.xy = replay.xy
+        self.k = replay.k
+        self.end_pos = 0
+        self.start_pos = 0
+        self.mods = replay.mods.short_name()
```

Comment:
just store `replay.mods`, you can pull out the `short_name` representation later but you also use the circlecore `Mod` object - currently by accessing `player.replay`, but ideally `player` shouldn't store `replay` at all. Any information needed should be made an instance attribute and `replay` should be left without a reference to be gc'd.